tests/profiles-enabled/test_imports.py

```diff
@@ -0,0 +1,320 @@
+            self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+            data = response.json()['data']
+            # upload_status field is not configurable at creation
+            self.assertEqual(data['attributes']['upload_status'], 'PENDING')
+            self.assertEqual(data['attributes']['processing_status'], 'PENDING')
+            # Wallets and storage should be present in response
+            self.assertEqual(data['attributes'].get('storage_credentials_id'), STORAGE_CRED_ID)
+            self.assertEqual(data['attributes'].get('shipper_wallet_id'), SHIPPER_WALLET_ID)
+            self.assertEqual(data['attributes'].get('carrier_wallet_id'), CARRIER_WALLET_ID)
+            xls_obj = ShipmentImport.objects.get(id=data['id'])
+            fields = data['meta']['presigned_s3']['fields']
+
+            # xls file upload
+            put_url = data['meta']['presigned_s3']['url']
+            with open(xls_path, 'rb') as xls:
+                res = requests.post(put_url, data=fields, files={'file': xls})
+
+            self.assertEqual(res.status_code, status.HTTP_204_NO_CONTENT)
+            self.assertTrue(self.s3_object_exists(xls_obj.s3_key))
+
+            # --------------------- Upload xlsx file --------------------#
+            response = self.client.post(url, xlsx_file_data, format='json')
+            self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+            data = response.json()['data']
+            self.assertEqual(data['attributes']['upload_status'], 'PENDING')
+            # processing_status field is not configurable at creation
+            self.assertEqual(data['attributes']['processing_status'], 'PENDING')
+            xlsx_obj = ShipmentImport.objects.get(id=data['id'])
+            fields = data['meta']['presigned_s3']['fields']
+
+            # xlsx file upload
+            put_url = data['meta']['presigned_s3']['url']
+            with open(xlsx_path, 'rb') as xlsx:
+                res = requests.post(put_url, data=fields, files={'file': xlsx})
+
+            self.assertEqual(res.status_code, status.HTTP_204_NO_CONTENT)
+            self.assertTrue(self.s3_object_exists(xlsx_obj.s3_key))
+
+            # -------------------- test patch object -------------------#
+            patch_csv_data = {
+                'name': 'Can update name',
+                'file_type': 'XLS',     # Cannot be modified
+                'owner_id': 'new_owner_id',     # Cannot be modified
+                'masquerade_id': 'new_masquerade_id',     # Cannot be modified
+                "upload_status": "Complete",
+                "processing_status": "Complete",
+                "report": {
+                    "success": 15,
+                    "failed": 0
+                }
+            }
+            csv_patch_url = f'{url}/{csv_obj.id}/'
+            response = self.client.patch(csv_patch_url, data=patch_csv_data, format='json')
```

Comment:
This test is for updating instead of posting right? I think it might be nice to separate these tests out into their own functions for greater clarity, especially if trying to test a change to a specific endpoint. It's a LOT easier to do if you model it after the tests in the shipment states branch.